sources/packages/backend/libs/services/src/disbursement-overaward/disbursement-overaward-service.ts

```diff
@@ -0,0 +1,53 @@
+import { Injectable } from "@nestjs/common";
+import { DataSource, EntityManager } from "typeorm";
+import { RecordDataModelService, DisbursementOveraward } from "@sims/sims-db";
+import {
+  AwardOverawardBalance,
+  StudentOverawardBalance,
+} from "./disbursement-overaward.models";
+
+/**
+ * Service layer for Student Application disbursement schedules.
+ */
+@Injectable()
+export class DisbursementOverawardService extends RecordDataModelService<DisbursementOveraward> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(DisbursementOveraward));
+  }
+
+  /**
+   * Sum the total overawards per value code (e.g. CSLF, BCSL) for the student.
+   * @param studentId student to get the balance.
+   * @param entityManager used to execute the queries in the same transaction.
+   * @returns the sum of the overawards grouped by the award type and them by
+   * the student id.
+   */
+  async getOverawardBalance(
+    studentIds: number[],
+    entityManager: EntityManager,
+  ): Promise<StudentOverawardBalance> {
+    // This query supports up to 65000 students.
+    const distinctStudentIds = [...new Set(studentIds)];
+    const totalAwards = await entityManager
+      .getRepository(DisbursementOveraward)
+      .createQueryBuilder("disbursementOveraward")
+      .select("student.id", "studentId")
+      .addSelect("disbursementOveraward.disbursementValueCode", "valueCode")
+      .addSelect("SUM(disbursementOveraward.overawardValue)", "total")
+      .innerJoin("disbursementOveraward.student", "student")
+      .where("student.id IN (:...studentIds)", {
+        studentIds: distinctStudentIds,
+      })
+      .groupBy("student.id")
+      .addGroupBy("disbursementOveraward.disbursementValueCode")
+      .getRawMany<{ studentId: number; valueCode: string; total: number }>();
```

Comment:
will this select only these fields `{ studentId: number; valueCode: string; total: number }`?

Reply:
Not sure if I got the question, please see the below image. The object type mentioned in the comment was added just to provide a temporary typing to allow the transformation to the `StudentOverawardBalance`. Please let me know if further explanation is needed.
![image](https://user-images.githubusercontent.com/61259237/211438638-6bc30c2e-3be4-4415-a7ab-bf4d76291ecd.png)